### PR TITLE
Finding Search API

### PR DESCRIPTION
**Pull request by lezzago**
*Issue #, if available:*
N/A
*Description of changes:*
- Supports a new Findings Search API to better search the findings and attach its corresponding documents
*CheckList:*
[X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/src/main/kotlin/org/opensearch/alerting/model/FindingWithDocs.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        builder.startObject()
+            .field(FINDING_FIELD, finding)
+            .field(DOCUMENTS_FIELD, documents)
+        builder.endObject()
+        return builder
+    }
+
+    companion object {
+        const val FINDING_FIELD = "finding"
+        const val DOCUMENTS_FIELD = "documents"
```
Comment (getsaurabh02) [inline alerting/src/main/kotlin/org/opensearch/alerting/model/FindingWithDocs.kt:44]: nit: change to ```document_list```

File: alerting/src/main/kotlin/org/opensearch/alerting/resthandler/RestGetFindingsSearchAction.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * This class consists of the REST handler to search findings .
+ */
+class RestGetFindingsSearchAction : BaseRestHandler() {
+
+    private val log = LogManager.getLogger(RestGetFindingsSearchAction::class.java)
+
+    override fun getName(): String {
+        return "get_findings_search_action"
```
Comment (getsaurabh02) [inline alerting/src/main/kotlin/org/opensearch/alerting/resthandler/RestGetFindingsSearchAction.kt:32]: fix name?

File: alerting/src/main/kotlin/org/opensearch/alerting/AlertingPlugin.kt [hunk 3]
```diff
@@ -178,7 +182,8 @@ internal class AlertingPlugin : PainlessExtension, ActionPlugin, ScriptPlugin, R
             RestSearchEmailGroupAction(),
             RestGetEmailGroupAction(),
             RestGetDestinationsAction(),
-            RestGetAlertsAction()
+            RestGetAlertsAction(),
+            RestGetFindingsSearchAction()
```
[the remark this answers is not in the document]
Comment (lezzago) [reply alerting/src/main/kotlin/org/opensearch/alerting/AlertingPlugin.kt:186]: I am renaming to `RestGetFindingsAction` as this keeps in more consistent to `GetAlerts` and `GetDestinations` APIs which are more similar to this API

File: alerting/src/main/kotlin/org/opensearch/alerting/action/GetFindingsSearchRequest.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import org.opensearch.action.ActionRequestValidationException
+import org.opensearch.alerting.model.Table
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.search.fetch.subphase.FetchSourceContext
+import java.io.IOException
+
+class GetFindingsSearchRequest : ActionRequest {
+    val findingId: String?
+    val version: Long
```
[the remark this answers is not in the document]
Comment (lezzago) [reply alerting/src/main/kotlin/org/opensearch/alerting/action/GetFindingsSearchRequest.kt:18]: removing

File: alerting/src/main/kotlin/org/opensearch/alerting/model/Finding.kt [hunk 5]
```diff
@@ -100,22 +101,24 @@ class Finding(
...
             lateinit var relatedDocId: String
             lateinit var monitorId: String
             lateinit var monitorName: String
             lateinit var index: String
             val queries: MutableList<DocLevelQuery> = mutableListOf()
             lateinit var timestamp: Instant
-            lateinit var triggerId: String
-            lateinit var triggerName: String
+            var triggerId: String? = null
+            var triggerName: String? = null
```
[the remark this answers is not in the document]
Comment (lezzago) [reply alerting/src/main/kotlin/org/opensearch/alerting/model/Finding.kt:113]: removing this

**Review by getsaurabh02 — APPROVED** [review]

**Pull request merged**